tuf/scheme/ecdsa_sha256.go

```diff
@@ -37,15 +50,124 @@ func NewECDSA() *ECDSA {
 	}
 	return &ECDSA{
 		PrivateKey: privateKey,
-		PublicKey:  privateKey.Public(),
-		KeyType:    ECDSA256,
+		KeyType:    types.ECDSA256,
 	}
 }
 
-func (ecdsa *ECDSA) Store(filename string) error {
-	keyJson, err := json.Marshal(ecdsa)
+func (ecdsaKey *ECDSA) encode() (string, string, error) {
+	x509Encoded, err := x509.MarshalECPrivateKey(ecdsaKey.PrivateKey)
+	if err != nil {
+		return "", "", err
+	}
+	pemEncoded := pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: x509Encoded})
+
+	x509EncodedPub, err := x509.MarshalPKIXPublicKey(&ecdsaKey.PrivateKey.PublicKey)
+	if err != nil {
+		return "", "", err
+	}
+	pemEncodedPub := pem.EncodeToMemory(&pem.Block{Type: "PUBLIC KEY", Bytes: x509EncodedPub})
+	return string(pemEncoded), string(pemEncodedPub), nil
+}
+
+func (ecdsaKey *ECDSA) decode(pemEncoded string) error {
+	block, _ := pem.Decode([]byte(pemEncoded))
+	privateKey, err := x509.ParseECPrivateKey(block.Bytes)
 	if err != nil {
 		return err
 	}
-	return ioutil.WriteFile(filename, keyJson, 0644)
+	ecdsaKey.PrivateKey = privateKey
+	ecdsaKey.KeyType = types.ECDSA256
+	return nil
+}
+
+func (ecdsaKey *ECDSA) Store(filename string) error {
+	privateKey, publicKey, err := ecdsaKey.encode()
+	schemeKey := SchemeKey{
+		PrivateKey: privateKey,
+		PublicKey:  publicKey,
+		KeyType:    ecdsaKey.KeyType,
+	}
+	jsonBytes, err := json.Marshal(schemeKey)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(filename, jsonBytes, 0644)
+}
+
+func (ecdsaKey *ECDSA) Sign(singedMetadata interface{}) (string, error) {
+	// Convert singedMetadata to bytes.
+	var buf bytes.Buffer
+	err := gob.NewEncoder(&buf).Encode(singedMetadata)
```

Comment:
Would it make sense to marshal to json instead of gob here?

Reply:
done